### DISCUSSION TITLE
Feature: Perform a warning kick before banning for spam

### DISCUSSION THREAD

**Discussion by gg4goat** [GriefPrevention/GriefPrevention#1197] [Ideas]
**My issue**
Sometimes regular players are being banned for spam on my server. Usually while spamming their friend with /w. Although I agree this shouldn't be allowed, It feels a little brutal since it's happing quite fast and the warning message could be missed.

**A possible solution**
Adding a warning kick before the spam ban with a message of the type "Spam is forbidden: next time you'll be banned".

Could save some moderation time.

**Comment by RoboMWM** [comment]
yea, spam is something that will be revisited in v18 or so. Not sure if it'll get to that degree of control, but definitely will add more controls.
